### PR TITLE
Add source_url and update pipeline [semver:patch]

### PR DESCRIPTION
**Pull request by lokst**
The only change to the orb is to add `source_url`.
The other changes in this PR are to update the pipeline to be in line with Orb Starter Kit's, and to update the readme.

### CODE REVIEW

File: src/@orb.yml [hunk 1]
```diff
@@ -2,7 +2,9 @@ version: 2.1
 
 description: >
   Build images and push them to the Amazon Elastic Container Registry.
-  See this orb's source: https://github.com/circleci-public/aws-ecr-orb
+
+display:
+  source_url: https://github.com/CircleCI-Public/aws-ecr-orb
```
Comment (KyleTryon) [inline src/@orb.yml]: How would we feel about adding: 
`home_url: https://aws.amazon.com/ecr/`

Comment (lokst) [reply src/@orb.yml]: @KyleTryon I think it makes sense to add it if this orb was made in partnership with AWS (I'm not sure about its origins), but if not, I'm not too sure what our policy should be. What do you think?

Comment (KyleTryon) [reply src/@orb.yml]: AWS is a bit of a special case. We do have somewhat of a formal partnership with them even though they do not actually author their own orbs. I prefer it for consistency too.

Comment (lokst) [reply src/@orb.yml]: @KyleTryon Thanks for clarifying! I agree that we should add it in our partner orbs. I went ahead and merged this one, but we can add `home_url` in next time!

**Pull request merged**